src/tests/adr/function_calls_adr.rs

```diff
@@ -0,0 +1,189 @@
+use crate::test_utils::tests::codegen;
+
+/// # Architecture Design Record: Function Calls
+///
+/// Function calls are handled differently compared to calls to Programs and FunctionBlocks. While
+/// most of the parameters of a Program/FB are part of its state, they are stored in a struct that
+/// is passed to it. Functions are stateless (between calls) and therefore get all their
+/// parameters handed one by one, exactly like we would expect it from a C-like language.
+#[test]
+fn function_calls_pass_single_arguments() {
```

Comment:
The functions are already covered in the pou_adr (Calling and returning complex types) maybe append/extend that?

Reply:
yes ill move it over

Reply:
all but one test were already covered by the POU_adr ... so I only moved the one that explains the `{ref}' behavior to the pou adr.